tests/unit/test_download.py

```diff
@@ -413,6 +413,50 @@ def test_unpack_file_url_thats_a_dir(self, tmpdir, data):
         assert os.path.isdir(os.path.join(self.build_dir, 'fspkg'))
 
 
+@pytest.mark.parametrize('exclude_dir', [
+    '.nox',
+    '.tox'
+])
+def test_unpack_file_url_with_excluded_dirs(exclude_dir):
+
+    def touch(path):
+        with open(path, 'a'):
+            os.utime(path, None)
+
+    src_dir = mkdtemp()
+    src_included_file = os.path.join(src_dir, 'file.txt')
+    src_excluded_dir = os.path.join(src_dir, exclude_dir)
+    src_excluded_file = os.path.join(src_dir, exclude_dir, 'file.txt')
+    src_included_dir = os.path.join(src_dir, 'subdir', exclude_dir)
+
+    # set up source directory
+    os.makedirs(src_excluded_dir, exist_ok=True)
+    os.makedirs(src_included_dir, exist_ok=True)
+    touch(src_included_file)
```

Comment:
I would use [`Path.touch()`](https://github.com/pypa/pip/blob/82dbcdae87f7a8aaaf1a9bae663a5fac8c856a40/tests/lib/path.py#L206) after switching to `tmpdir`.